### PR TITLE
Use a stream when uploading database contents

### PR DESCRIPTION
**Pull request by robertbrignull**
Closes https://github.com/github/codeql-action/issues/1383

Instead of loading the entire database contents into memory, pass a `ReadStream` to `request`. Internally, octokit uses `node-fetch` and that is capable of accepting a stream. I then had to add the content length manually because the database upload API requires this header, but it wasn't populated when passing a stream to `request`.

Finally, I moved slightly more logic inside the try-catch. This should mean that if the database upload fails again for a user it won't block the main analysis and generally won't be an issue.

### Merge / deployment checklist

- [x] Confirm this change is backwards compatible with existing workflows.
- [x] Confirm the [readme](https://github.com/github/codeql-action/blob/main/README.md) has been updated if necessary.
- [x] Confirm the [changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md) has been updated if necessary.

### CODE REVIEW

**Comment by robertbrignull** [comment]
I tested this in [another repo](https://github.com/robertbrignull-test-org/pub/actions/runs/3856209004/jobs/6572369893) and the database upload was successful.

**Comment by angelapwen** [comment]
Hm, looks like there's a genuine unit test failure related to the Windows file system 👁️

**Comment by aeisenberg** [comment]
Looks reasonable to me. @angelapwen is right. I wonder if you need to explicitly close the read stream before you can delete it.

File: src/database-upload.ts [hunk 1]
```diff
@@ -36,29 +36,34 @@
...
+            repo: repositoryNwo.repo,
+            language,
+            name: `${language}-database`,
+            data: bundledDbReadStream,
+            headers: {
+              authorization: `token ${apiDetails.auth}`,
+              "Content-Type": "application/zip",
+              "Content-Length": bundledDbSize,
+            },
+          }
```
[the remark this answers is not in the document]
Comment (robertbrignull) [reply src/database-upload.ts:61]: This is intended and not changing behaviour in this PR

**Comment by robertbrignull** [comment]
Thanks @angelapwen and @aeisenberg. I've added a `finally` block to close the stream. I'm glad we have those windows unit tests to catch this!

**Review by elenatanasoiu — APPROVED** [review]

**Pull request merged**